test/unit/plugins/validate-semantic/2and3/schemas.js

```diff
@@ -169,3 +169,312 @@ describe(`validation plugin - semantic - 2and3 schemas`, () => {
+                schema: {
+                  type: "number",
+                  enum: [1, "2", 3]
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      const firstError = allErrors[0]
+      expect(allErrors.length).toEqual(1)
+      expect(firstError.level).toEqual("warning")
+      expect(firstError.message).toEqual("enum value should conform to its schema's `type`")
+    })
+  })
+  
+  it("should not return an error when all items are number in a enum number type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "number",
+                in: "query",
+                schema: {
+                  type: "number",
+                  enum: [1, 2, 3]
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      expect(allErrors.length).toEqual(0)
+    })
+  })
+
+  //Array Tests
+
+  it("should return an error for a non array value in a enum array type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "arraySample",
+                in: "query",
+                schema: {
+                  type: "array",
+                  enum: [1, 2, 3]                  
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      const secondError = allErrors[1]
+      expect(allErrors.length).toEqual(2)
+      expect(secondError.level).toEqual("warning")
+      expect(secondError.message).toEqual("enum value should conform to its schema's `type`")
+      expect(secondError.level).toEqual("warning")
+      expect(secondError.message).toEqual("enum value should conform to its schema's `type`")
+    })
+  })
+  
+  it("should not return an error when all items are array in a enum array type in OpenApi 3", () => {
+    const spec = {
+      openapi: "3.0.0",
+      "paths": {
+        "/pets": {
+          "get": {
+            "parameters": [
+              {
+                name: "arraySample",
+                in: "query",
+                schema: {
+                  type: "array",
+                  enum: [[1,2],[3,4]]
+                }
+              },
+            ]
+          }
+        }
+      }
+    }
+
+    return validateHelper(spec)
+    .then(system => {
+      const allErrors = system.errSelectors.allErrors().toJS()
+      expect(allErrors.length).toEqual(1)
```

Comment:
What's the error you're seeing? You can filter out unrelated ones, [here's](https://github.com/swagger-api/swagger-editor/blob/724fac76bf88016bbee397fd8e1a5517f5591afc/test/unit/plugins/validate-semantic/2and3/refs.js#L453-L454) an example of how to.

I'd like to avoid baking in the single failing error, because choosing this "magic number" makes this test case sensitive to changes outside of the code it's testing. We can consciously choose to keep it if it's a pain to work around, though.

Reply:
The error has been fixed, it was related to items sibling element when using array types. Changed the test case for this purpose-